src/main/java/frc/robot/commands/gpm/IntakeNote.java

```diff
@@ -0,0 +1,38 @@
+package frc.robot.commands.gpm;
+
+import edu.wpi.first.math.filter.Debouncer;
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.constants.IntakeConstants;
+import frc.robot.subsystems.gpm.Intake;
+
+public class IntakeNote extends Command {
+
+    private final Intake intake;
+    private final Debouncer stallDebouncer;
+
+
+    public IntakeNote(Intake intake) {
+        addRequirements(intake);
+        this.intake = intake;
+        this.stallDebouncer = new Debouncer(IntakeConstants.INTAKE_STALL_TIME, Debouncer.DebounceType.kBoth);
+    }
+
+    @Override
+    public void initialize() {
+        intake.setMode(Intake.Mode.INTAKE);
+    }
+
+    // TODO: This code might need some adjusting, as the cone passes through the intake.
+    @Override
+    public boolean isFinished() {
+        return stallDebouncer.calculate(
+                intake.getCurrent() >= IntakeConstants.INTAKE_CURRENT_STOP
+        );
```

Comment:
This command probably needs the indexer (should be merged soon) as well as the intake. Because it's continuous, we don't want to stop it when it has a higher current or a note in it. It also might not have a high current long enough to trigger the debouncer. Using the indexer's hasNote() would be better.

Reply:
done